lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -2357,16 +2357,24 @@ Scalar DistributionImplementation::computeScalarQuantile(const Scalar prob,
   LOGDEBUG(OSS() << "DistributionImplementation::computeScalarQuantile: lower=" << lower << ", upper=" << upper);
   if (prob < 0.0) return (tail ? upper : lower);
   if (prob >= 1.0) return (tail ? lower : upper);
-  const Scalar q = tail ? 1.0 - prob : prob;
+  const Scalar p = tail ? 1.0 - prob : prob;
   const CDFWrapper wrapper(this);
   const Function f(bindMethod<CDFWrapper, Point, Point>(wrapper, &CDFWrapper::computeCDF, 1, 1));
   const Scalar leftTau = (std::isinf(lower) ? -SpecFunc::ActualMaxScalar : lower);
   const Scalar leftCDF = 0.0;
   const Scalar rightTau = (std::isinf(upper) ? SpecFunc::ActualMaxScalar : upper);
   const Scalar rightCDF = 1.0;
   Brent solver(quantileEpsilon_, cdfEpsilon_, cdfEpsilon_, quantileIterations_);
-  const Scalar root = solver.solve(f, q, leftTau, rightTau, leftCDF, rightCDF);
+  Scalar root = solver.solve(f, p, leftTau, rightTau, leftCDF, rightCDF);
   LOGDEBUG(OSS() << "root=" << root);
+
+  // special case non strictly increasing CDF: retain the inf of the interval veryfing F(x)=p
+  if (computeCDF(root - quantileEpsilon_) >= prob)
+  {
+    solver.setResidualError(0.5 * cdfEpsilon_);
+    root = solver.solve(f, p - cdfEpsilon_, leftTau, root, leftCDF, prob);
+    LOGDEBUG(OSS() << "inf root=" << root);
```

Comment:
At this point you can add a 1st order correction. You solved $F(x_p)=p-\epsilon$, and you are looking for $\eta$ such that $F(x_p+\eta)=p$. But, up to the first order, $F(x_p+\eta)=F(x_p)+\eta F'(x_p)$, so $p=p-\epsilon+\eta F'(x_p)$, resulting into $\eta=\epsilon/F'(x_p)$. It works only if $F'(x_p)\neq 0$, which is the case for a continuous mixture but not for a discrete distribution or a mix of discrete and continuous distributions.

Reply:
lets continue this in #2564